rotkehlchen/db/filtering.py

```diff
@@ -580,6 +582,25 @@ class DBMultiIntegerFilter(DBMultiValueFilter[int]):
     """Filter a column having an integer value out of a selection of values"""
 
 
+@dataclass(init=True, repr=True, eq=True, order=False, unsafe_hash=False, frozen=False)
+class DBOptionalChainAddressesFilter(DBFilter):
+    """Filter the address column by a selection of optional chain addresses"""
+    optional_chain_addresses: Optional[list[OptionalChainAddress]]
+
+    def prepare(self) -> tuple[list[str], list[str]]:
+        query_filters = []
+        bindings: list[Union[str, ChecksumEvmAddress]] = []
+        if self.optional_chain_addresses is not None:
+            for optional_chain_address in self.optional_chain_addresses:
+                query_part = 'address = ?'
```

Comment:
This is a weird way of doing it. Instead I want you to use already existing functionality for filtering for a chain of addresses.

Check `DBTransactionsPendingDecodingFilter` in this same file: https://github.com/rotki/rotki/blob/2b358c20a052729238f79c24733d88bc8448e6d9/rotkehlchen/db/filtering.py#L174

And instead abstract that logic and move it here. And then use this `DBOptionalChainAddressesFilter` in `DBTransactionsPendingDecodingFilter` through inheritance as is done in other places in this file.

Reply:
They two filters are very different, and check different fields

Reply:
Discussed this in a call and this is indeed two different things as this is using:

```python
class OptionalChainAddress(NamedTuple):
    address: ChecksumAddress
    blockchain: Optional[SupportedBlockchain]
    ```

My comment is invalid.